app/stac_api/management/commands/calculate_extent.py

```diff
@@ -0,0 +1,108 @@
+                    self.print_success("No collections updated")
+                    return
+
+            start = time.monotonic()
+            for collection_id in collections:
+                cursor.execute(
+                    """
+                    -- Compute collection extent
+                    WITH collection_extent AS (
+                        SELECT
+                            item.collection_id,
+                            ST_SetSRID(ST_EXTENT(item.geometry),4326) as extent_geometry,
+                            MIN(LEAST(item.properties_datetime, item.properties_start_datetime))
+                                as extent_start_datetime,
+                            MAX(GREATEST(item.properties_datetime, item.properties_end_datetime))
+                                as extent_end_datetime
+                        FROM stac_api_item AS item
+                        WHERE item.collection_id = %s
+                        GROUP BY item.collection_id
+                    UNION
+                        -- This covers the case that the last item of a collection is deleted.
+                        SELECT %s AS collection_id, NULL, NULL, NULL
+                    LIMIT 1
+                    )
+                    -- Update related collection extent
+                    UPDATE stac_api_collection SET
+                        extent_out_of_sync = FALSE,
+                        extent_geometry = collection_extent.extent_geometry,
+                        extent_start_datetime = collection_extent.extent_start_datetime,
+                        extent_end_datetime = collection_extent.extent_end_datetime
+                    FROM collection_extent
+                    WHERE id = collection_extent.collection_id;
+                    """, [collection_id[0], collection_id[0]]
+                )
+                self.print_success(f"collection.id={collection_id[0]} extent updated.")
+            self.print_success(
+                f"successfully updated extent of {len(collections)} collections in " +
+                f"{(time.monotonic()-start):.3f}s"
+            )
+
+
+class Command(BaseCommand):
+    help = """Calculate the collaction spacial and temporal extent for all collections that have
+    'extent_out_of_sync' set to true. After update, 'extent_out_of_sync' will be set to False.
+    This command is thought to be scheduled as cron job.
+    """
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            '-a',
+            '--all',
+            type=bool,
+            nargs='?',
+            default=False,
+            help='Update extent for all collections'
+        )
+        parser.add_argument(
+            '-f',
+            '--force',
+            type=bool,
+            nargs='?',
+            default=False,
+            help='Run all without confirmation'
+        )
+
+    def handle(self, *args, **options):
+        # if flag is passed without argument value is set to none
```

Comment:
I think you might be able to simplify this by adding `action='store_true'` to the add_argument of these fields. See https://docs.python.org/3/library/argparse.html#action

Reply:
Thanks, this is exactly what I was looking for.